lms/static/scripts/frontend_apps/components/test/BasicLtiLaunchApp-test.js

```diff
@@ -433,6 +455,57 @@ describe('BasicLtiLaunchApp', () => {
       await waitForElement(wrapper, 'FakeDialog[title="Authorize Hypothesis"]');
       await contentHidden(wrapper);
     });
+
+    context('when auth fails multiple times in a row', () => {
+      it('shows an error dialog if the first requests rejects and second also rejects', async () => {
+        const wrapper = renderLtiLaunchApp();
+        // Both requests reject first
+        contentUrlReject(new ApiError(400, {}));
+        groupsCallReject(new ApiError(400, {}));
+
+        const authButton = await waitForElement(
+          wrapper,
+          'Button[label="Authorize"]'
+        );
+
+        // Click the "Authorize" button.
+        resetApiCalls();
+        // contentUrlReject rejects again
+        contentUrlReject(new ApiError(400, {}));
+        groupsCallReject(new ApiError(400, {}));
+        // Error prompt should remain.
+        authButton.prop('onClick')();
+        await waitForElement(
+          wrapper,
+          'FakeDialog[title="Authorize Hypothesis"]'
+        );
+      });
+      it('shows an error dialog if contentUrl succeeds but groups rejects first', async () => {
+        const wrapper = renderLtiLaunchApp();
+        // Both requests reject first
+        contentUrlReject(new ApiError(400, {}));
+        groupsCallReject(new ApiError(400, {}));
+
+        const authButton = await waitForElement(
+          wrapper,
+          'Button[label="Authorize"]'
+        );
+
+        // Click the "Authorize" button.
+        resetApiCalls();
+        // groups still fails, but contentUrl does not.
+        groupsCallReject(new ApiError(400, {}));
+        contentUrlResolve({
+          via_url: 'https://via.hypothes.is/123',
+        });
+        // Error prompt should remain.
+        authButton.prop('onClick')();
+        await waitForElement(
+          wrapper,
+          'FakeDialog[title="Authorize Hypothesis"]'
```

Comment:
This test passes if run against the version of `BasicLtiLaunchApp.js` from the `master` branch. I guess that is happening because the "Authorize" dialog is still visible immediately after the "Authorize" button is clicked.